dissect/cstruct/types/flag.py

```diff
@@ -1,15 +1,18 @@
 from __future__ import annotations
 
-from typing import BinaryIO, List, Tuple, Union
+from enum import IntFlag
 
-from dissect.cstruct.types import Enum, EnumInstance
+from dissect.cstruct.types.base import BaseType
+from dissect.cstruct.types.enum import EnumMetaType
 
 
-class Flag(Enum):
-    """Implements a Flag type.
+class Flag(BaseType, IntFlag, metaclass=EnumMetaType):
```

Comment:
```suggestion
class Flag(IntFlag, BaseType, metaclass=EnumMetaType):
```

Is there a reason the first base class is not the Python built-in class here? All other cstruct types have the Python built-in as the first base class.